src/VisualStudio/IntegrationTest/TestUtilities/InProcess/VisualStudioWorkspace_InProc.cs

```diff
@@ -178,5 +178,24 @@ public string GetWorkingFolder()
             var service = _visualStudioWorkspace.Services.GetRequiredService<IPersistentStorageLocationService>();
             return service.TryGetStorageLocation(_visualStudioWorkspace.CurrentSolution.Id);
         }
+
+        private VisualStudioWorkspaceChangeWaiter_InProc _workspaceChangeWaiter;
```

Comment:
I was hoping to not have this field and instead return the `VisualStudioWorkspaceChangeWaiter_InProc` from BeginWaitForWorkspaceChange, and then have the code in EndWaitForWorkspaceChange live in the out-of-proc portion, but something is breaking with Remoting.